hatchways/team-old-fashioned
# Feature message notifications

### What this PR does (required):
- Adds notifications for messages
- Adds mark-as-read functionality to the front-end
- Links notification to the relevant contest or to the Messages page

### Screenshots / Videos (required):
https://user-images.githubusercontent.com/78225194/129843768-9a46d824-bbe9-4f72-ba5e-0f37f0b4c00d.mp4

### Any information needed to test this feature (required):
Open two accounts simultaneously.
- To test message notifications:
-- Send messages to one another.
-- Check whether Message notifications appear in the Messages dropdown.
-- Click the notification which should route to the Messages page and remove the notification from the dropdown
- To test submission notifications:
-- Submit to a contest.
-- In the contest owner profile, check that a notification has been made for the submission.
-- Click on the notification. This should route to the contest page and remove the notification from the Notifications dropdown.
-- Go to the Notifications page. The clicked notification should be displayed still but without the indicator for unread messages.

### Any issues with the current functionality (optional):
- Client is not receiving emitted confirmation from the server that the notification has been read.
- Conversations are not yet set up on the front end to be accessible by ID so message notification link routes to Messages page only.
- Possible improvement: Notifications can be condensed to one notification per contest-submitter combination or one notification per message sender.
- Possible additional feature: Notifications for contest deadline and winner selection.
